src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1316,14 +1346,18 @@ private static TimeSeriesItem CreateTimeSeriesItem(DocumentsOperationContext con
             var segmentPtr = reader.Read((int)TimeSeriesTable.Segment, out int segmentSize);
             var keyPtr = reader.Read((int)TimeSeriesTable.TimeSeriesKey, out int keySize);
 
+            var key = new LazyStringValue(null, keyPtr, keySize, context);
+
             TimeSeriesValuesSegment.ParseTimeSeriesKey(keyPtr, keySize, out var docId, out var name, out var baseline);
 
             return new TimeSeriesItem
```

Comment:
TODO:
we need to introduce proper time series container like `Document` and use instead of `string properties, `LazyStringValue` to avoid allocations

Reply:
done